modules/compiler/spirv-ll/source/builder_debug_info.cpp

```diff
@@ -0,0 +1,44 @@
+
+// Copyright (C) Codeplay Software Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License") with LLVM
+// Exceptions; you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://github.com/codeplaysoftware/oneapi-construction-kit/blob/main/LICENSE.txt
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <spirv-ll/builder_debug_info.h>
+
+namespace spirv_ll {
+
+llvm::Error DebugInfoBuilder::create(OpExtInst const &) {
```

Comment:
For `OpNop`, we return type `cargo::optional<Error>`, value `cargo::nullopt`. Since we are essentially treating it the same, we should probably to the same, so that the caller can assume that it only receives an `llvm::Error` if an error occurred?

Reply:
Are you perhaps looking at older code? I recently changed everything to `llvm::Error`, so `OpNop` now returns `llvm::Error::success()` too.

Reply:
Oh! You changed that in #218, which I didn't have locally yet. In that case, looks good to me.